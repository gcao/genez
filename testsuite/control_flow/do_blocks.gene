#!/usr/bin/env gene run

# Test: Do blocks for sequential execution
(do
  (print "First statement")
  (print "Second statement")
  (print "Third statement"))

# Do block with variable scope
(do
  (var x 42)
  (print "Inside do block: x = " x))