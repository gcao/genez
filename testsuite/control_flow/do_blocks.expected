First statement
Second statement
Third statement
Inside do block: x =  42